weldx/tests/test_utility.py

```diff
@@ -498,3 +501,137 @@ def test_xr_time_ref():
 
     da2.weldx.time_ref = t0
     assert da2.time.attrs["time_ref"] == t0
+
+
+def test_non_sense():
+    with pytest.raises(TypeError):
+        ut.compare_nested("asdf", "foo")
+
+    with pytest.raises(TypeError):
+        ut.compare_nested(b"asdf", b"foo")
+
+    assert not ut.compare_nested((1, 2, 3), {"f": 0})
+
+    assert not ut.compare_nested((1, 2, 3), "bar")
+
+
+def test_flat():
+    a = (1, 2, 3)
+    b = list(a)
+    assert ut.compare_nested(a, b)
+    b[-1] = 0
+    assert not ut.compare_nested(a, b)
+
+
+def test_compare_nested_diff_keys():
+    a = {
+        "blah": np.arange(3),
+    }
+    b = {
+        "blub": np.arange(3),
+    }
+    assert not ut.compare_nested(a, b)
+
+
+def test_compare_nested_dicts_same_elements_diff_content():
+    a = {
+        "blah": np.arange(3),
+        "x": {
+            0: [1, 2, 3],
+        },
+        "blub": False,
+    }
+    b = {
+        "blah": np.arange(3),
+        "x": {
+            0: [1, 2, 3, 4],
+        },
+        "blub": False,
+    }
+    assert not ut.compare_nested(a, b)
+
+
+def test_compare_nested_dicts_diff_content():
+    a = {
+        "blah": np.arange(3),
+        "x": {
+            0: [1, 2, 3],
+        },
+        "blub": False,
+    }
+    b = {
+        "blah": np.arange(3),
+        "x": {
+            0: [1, 2, 3],
+        },
+        "blub": True,  # DIFF!
+    }
+    assert not ut.compare_nested(a, b)
+
+
+def test_compare_nested_dicts_same_content():
+    a = {
+        "blah": np.arange(3),
+        "x": {
+            0: [1, 2, 3],
+        },
+        "blub": True,
+    }
+    b = dict(a)
+    assert ut.compare_nested(a, b)
+
+
+class TestWeldxExampleCompareNested(unittest.TestCase):
```

Comment:
what is the reason fpr using `unittest.TestCase`?
does it play nice with pytest?

Reply:
it does play along with pytest nicely. It is used to load the data only once. I could have used a pytest.fixture as well, but I'm more used to this pattern from the past. If you want I can rewrite it using pytest fixtures only.

Reply:
ok sounds nice, maybe you can give a quick intro into this style soon. :)

Reply:
I can do this on our jour fixe, like 15 minutes should be sufficient to get the full idea since all of us are familiar with unit testing, right?

Reply:
great 👍